packages/core/utils/src/BitSet.js

```diff
@@ -97,30 +89,102 @@ export class BitSet<Item> {
     return new BitSet({
       lookup: this._lookup,
       items: this._items,
-      initial: this._value,
+      initial: this._value.clone(),
     });
   }
 
   values(): Array<Item> {
     let values = [];
-    let tmpValue = this._value;
-    let i;
-
-    // This implementation is optimized for BitSets that contain a very small percentage
-    // of items compared to the total number of potential items. This makes sense for
-    // our bundler use-cases where Sets often contain <1% coverage of the total item count.
-    // In cases where Sets contain a larger percentage of the total items, a regular looping
-    // strategy would be more performant.
-    while (tmpValue > BIGINT_ZERO) {
-      // Get last set bit
-      i = tmpValue.toString(2).length - 1;
-
+    this._value.forEach(i => {
       values.push(this._items[i]);
+    });
+
+    return values;
+  }
+}
+
+// Small wasm program that exposes the `ctz` instruction.
+// https://developer.mozilla.org/en-US/docs/WebAssembly/Reference/Numeric/Count_trailing_zeros
+const wasmBuf = new Uint8Array([
+  0x00, 0x61, 0x73, 0x6d, 0x01, 0x00, 0x00, 0x00, 0x01, 0x06, 0x01, 0x60, 0x01,
+  0x7f, 0x01, 0x7f, 0x03, 0x02, 0x01, 0x00, 0x07, 0x0d, 0x01, 0x09, 0x74, 0x72,
+  0x61, 0x69, 0x6c, 0x69, 0x6e, 0x67, 0x30, 0x00, 0x00, 0x0a, 0x07, 0x01, 0x05,
+  0x00, 0x20, 0x00, 0x68, 0x0b, 0x00, 0x0f, 0x04, 0x6e, 0x61, 0x6d, 0x65, 0x02,
+  0x08, 0x01, 0x00, 0x01, 0x00, 0x03, 0x6e, 0x75, 0x6d,
+]);
+
+// eslint-disable-next-line
+const {trailing0} = new WebAssembly.Instance(new WebAssembly.Module(wasmBuf))
+  .exports;
+
+export class RawBitSet {
+  bits: Uint32Array;
+
+  constructor(maxBits: number) {
+    this.bits = new Uint32Array(Math.ceil(maxBits / 32));
+  }
+
+  clone(): RawBitSet {
+    let res = new RawBitSet(this.capacity);
+    res.bits.set(this.bits);
+    return res;
+  }
+
+  get capacity(): number {
+    return this.bits.length * 32;
+  }
+
+  add(bit: number) {
+    let i = bit >>> 5;
```

Comment:
Just curious, but why is this 5? lol

Reply:
Shifting right divides by powers of two, so this divides by 32. Bit shifting is faster than floating point division, and easier for the JS engine to optimize.